### PR TITLE
Merge GraphQL feature branch into master

### PR DESCRIPTION
**Pull request by edwardfeng-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/edwardfeng-db/mlflow/pull/10931?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10931/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10931
```

</p>
</details>

### Related Issues/PRs

 #xxx

https://github.com/mlflow/mlflow/pull/10474

### What changes are proposed in this pull request?

Adding GraphQL related features into MLFlow
- proto_to_graphql autogeneration
- `/graphql` handler in `handlers.py`
- Onboard `getRun` to graphql

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: requirements/test-requirements.txt [hunk 1]
```diff
@@ -44,7 +44,9 @@ databricks-sdk
 openai<1.0
 # Required for showing pytest stats
 psutil
+# Required for the GraphQL handler
+graphene<4
```
Comment (harupy) [inline requirements/test-requirements.txt:48]: I think `graphene` can be removed because it should be installed together with mlflow.

**Review by hubertzub-db — APPROVED** [review]
After some testing I confirm it's possible to get the run data (excluding experiment and model data yet) using calls to `/graphql`, approving from "UI" perspective

File: mlflow/server/graphql/autogenerated_graphql_schema.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+class MlflowParam(graphene.ObjectType):
+    key = graphene.String()
+    value = graphene.String()
+
+
+class MlflowMetric(graphene.ObjectType):
+    key = graphene.String()
+    value = graphene.Float()
+    timestamp = LongString()
+    step = LongString()
```
Comment (hubertzub-db) [inline mlflow/server/graphql/autogenerated_graphql_schema.py:55]: for compatibility with current REST API, can we retrieve those values as integers?

Comment (hubertzub-db) [reply mlflow/server/graphql/autogenerated_graphql_schema.py:55]: also it's int64 in protofiles: https://src.dev.databricks.com/mlflow/mlflow/-/blob/mlflow/protos/service.proto?L478-482

Comment (edwardfeng-db) [reply mlflow/server/graphql/autogenerated_graphql_schema.py:55]: Synced offline, this is for keeping consistency with the in house graphql schema we have

**Review by harupy — APPROVED** [review]

**Pull request merged**